nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -476,7 +476,7 @@ public Description matchAssignment(AssignmentTree tree, VisitorState state) {
       doUnboxingCheck(state, tree.getExpression());
     }
     // generics check
-    if (lhsType != null && lhsType.getTypeArguments().length() > 0) {
+    if (lhsType != null && lhsType.getTypeArguments().length() > 0 && config.isJSpecifyMode()) {
```

Comment:
@akulk022 I wrote these extra checks in NullAway.java some time back, but I don't think they are needed anymore.  Can you try removing all the changes in NullAway.java and see if the tests still pass?

Reply:
I tried removing the changes and saw some tests failing, I haven't looked into it more deeply yet, Should I do that?

Reply:
No need, I was just curious